docs/reference/openapi-rules.md

```diff
@@ -1,3 +1,546 @@
+**Bad Example**
+
+```yaml
+openapi: "3.0.2"
+info:
+  title: 'some title with eval(',
+```
+
+### no-script-tags-in-markdown
+
+This rule protects against a potential hack, for anyone bringing in description documents from third parties then generating HTML documentation. If one of those third parties does something shady like inject `<script>` tags, they could easily execute arbitrary code on your domain, which if it's the same as your main application could be all sorts of terrible.
+
+**Recommended:** Yes
+
+**Bad Example**
+
+```yaml
+openapi: "3.0.2"
+info:
+  title: 'some title with <script>alert("You are Hacked");</script>',
+```
+
+### openapi-tags-alphabetical
+
+OpenAPI object should have alphabetical `tags`. This will be sorted by the `name` property.
+
+**Recommended:** No
+
+**Bad Example**
+
+```yaml
+tags: 
+  - name: 'Badger'
+  - name: 'Aardvark'
+```
+
+**Good Example**
+
+```yaml
+tags: 
+  - name: 'Aardvark'
+  - name: 'Badger'
+```
+
+### openapi-tags
+
+OpenAPI object should have non-empty `tags` array.
+
+Why? Well, you _can_ reference tags arbitrarily in operations, and definition is optional...
+
+```
+/invoices/{id}/items:
+  get:
+    tags:
+    - Invoice Items
+```
+
+Defining tags allows you to add more information like a `description`. (See [tag-description](#tag-description))
+
+**Recommended:** No
+
+### operation-default-response
+
+**Recommended:** No
+
+### operation-description
+
+**Recommended:** Yes
+
+### operation-operationId
+
+This operation ID is essentially a reference for the operation, which can be used to visually suggest a connection to other operations. This is like some theoretical static HATEOAS-style referencing, but it's also used for the URL in some documentation systems.
+
+Make the value `lower-hyphen-case`, and try and think of a name for the action which does not relate to the HTTP message. Base it off the actual action being performed. `create-polygon`? `search-by-polygon`? `filter-companies`?
+
+**Recommended:** Yes
+
+### operation-operationId-valid-in-url
+
+Seeing as operationId is often used for unique URLs in documentation systems, it's a good idea to avoid non-URL safe characters.
+
+**Recommended:** Yes
+
+**Bad Example**
+
+```
+paths:
+  /pets:
+    get:
+      operationId: get cats
+```
+
+### operation-singular-tag
+
+Use just one tag for an operation, which is helpful for some documentation systems which use tags to avoid duplicate content.
+
+**Recommended:** No
+
+### operation-summary-formatted
+
+Operation `summary` should start with upper case and end with a dot.
+
+**Recommended:** No
+
+### operation-tags
+
+Operation should have non-empty `tags` array.
+
+**Recommended:** Yes
+
+### parameter-description
+
+Parameter objects should have a `description`.
+
+**Recommended:** No
+
+### path-declarations-must-exist
+
+Path parameter declarations cannot be empty, ex.`/given/{}` is invalid.
+
+**Recommended:** Yes
+
+### path-keys-no-trailing-slash
+
+**Recommended:** Yes
```

Comment:
```suggestion
Keep trailing slashes off of paths, as it can cause some confusion. Some web tooling (like mock servers, real servers, code generators, application frameworks, etc.) will treat `example.com/foo` and `example.com/foo/` as the same thing, but other tooling will not. Avoid any confusion by just documenting them without the slash, and maybe some tooling will let people shove a / on there when they're using it or maybe not, but at least the docs are suggesting how it should be done properly.

**Recommended:** Yes
```

Reply:
Thanks!